src/elementary_functions.jl

```diff
@@ -144,33 +149,36 @@ function polynomial_itensornetwork(
   g_tree = undirected_graph(random_bfs_tree(g, first(vertices(g))))
   s_tree = add_edges(rem_edges(_s, edges(g)), edges(g_tree))
   s_tree = IndsNetworkMap(s_tree, indexmap(s))
+  eltype = is_real(s) ? Float64 : ComplexF64
```

Comment:
This seems to assume C->C and R->R but we may have C->R

Reply:
The case I'm thinking of is a bit different. It would be an entirely real network which just takes complex numbers (think `f(z)=abs(z)` or `f(z) = real(z)`). In terms of flops/memory storage this would be nice to have the flexibility

Reply:
Hmmm this seems like a bit of a weird edge case to me? If the `itensornetwork` is encoding the values of a complex function then it should generically have complex elements? I'm not sure we should try to overcode to handle the case where the functions domain happens to be real.

Whilst such functions exist, really we shouldn't use a complex function at all to handle them.

E.g. `f(z) = abs(z)` should be viewed as `f(x,y) = sqrt(x*x + y*y)` and `f(z) = real(z)` should be viewed as `f(x,y) = x`

Reply:
As in like `Float64(itn::ITensorNetworkFunction)` which converts all the elements of the tensors to `Float64` etc? That seems like a good idea